proxy/http/HttpSM.cc

```diff
@@ -7408,11 +7406,10 @@ HttpSM::set_next_state()
       perform_cache_write_action();
       t_state.api_next_action = HttpTransact::SM_ACTION_API_SEND_RESPONSE_HDR;
 
-      // check to see if we are going to handle the redirection from server response and if there is a plugin hook set
-      if (hooks_set && is_redirect_required() == false) {
+      // check to see if there is a plugin hook set
+      if (hooks_set) {
         do_api_callout_internal();
       } else {
-        do_redirect();
```

Comment:
It looks good to me,

In the original code. we only invoke the hook after redirection . After this change, we will invoke the hook twice (3XX response and normal response ). So it does change the plugin behavior, doesn't it ? If so, I'd like to write a comment or document it .

Reply:
> So it does change the plugin behavior, doesn't it ?

Changing plugin behavior in this way is not the intention of this patch. I think I should take a look and see if the change can be made without having this effect, if possible.

Reply:
@scw00 Because this case is a cache hit, we will not be following redirects. `is_redirect_required()` should return `false`, and so we should only call `do_api_callout_internal()` when `hooks_set` is `true` and otherwise we call `handle_api_return()`.

So it should not be the case that we call hooks twice, but please point out if I missed something.

Reply:
I got it. Thanks for your explanation